homework-submission/todoapp/src/Stores/TodoStore.js

```diff
@@ -0,0 +1,53 @@
+import { observable, action } from 'mobx';
+import todoObj from '../data/todos.json';
+
+
+class TodosStore {
+  @observable todos = todoObj;
+
+  @action checkBoxHandler = (id) => {
+    const newTodos = this.todos.map(item => {
+      if (item.id === parseInt(id, 10)) {
+        return {
+          ...item,
+          done: !item.done
+        };
+      }
+      return item;
+    });
+    this.todos = newTodos;
+  }
+
+  @action deleteHandler = (ID) => {
+    const newTodos = this.todos.filter(item => item.id !== ID);
+    this.todos = newTodos;
+  }
+
+  @action editHandler = (ID) => {
+    const Index = this.todos.findIndex(todo => todo.id === ID)
+    const newTodo = { ...this.todos[Index], edit: !this.todos[Index].edit }
+    this.todos[Index] = newTodo
+  }
+
+  @action updateHandler = (oldDescription, newDescription, newDeadline) => {
```

Comment:
Hmm, In the `TodoElement` Component file line 14 this confused me a bit, it made the question: why you are updating and then editing 🤔.

So can `toggleUpdate` toggles the meaning of it's action `on` again 🙂 ?
I am sure you may have a better than this 😄